tests/ui/consts/const-mut-refs/const_mut_refs.rs

```diff
@@ -28,6 +30,10 @@ const fn bazz(foo: &mut Foo) -> usize {
     foo.x
 }
 
+// Empty slices get promoted so this passes the static checks.
+// Make sure it also passes the dynamic checks.
+static MUTABLE_REFERENCE_HOLDER: Mutex<&mut [u8]> = Mutex::new(&mut []);
```

Comment:
If I'm reading the code right, the following will now also be legal, right?

```rust
static mut FOO: &mut [u8] = &mut [42];
static BAR: Mutex<&mut [u8]> = Mutex::new(unsafe { &mut *FOO });
```

Reply:
edit: ah no, we're still using a ZST check on top

Reply:
I guess this is tested in mut_ref_in_final_dynamic_check

Reply:
I'll add this test in that file, just to be extra safe.

Reply:
Done that.